src/audio/codec_adapter/codec/generic.c

```diff
@@ -246,6 +246,22 @@ int codec_prepare(struct comp_dev *dev)
 	return ret;
 }
 
+static int process_passthrough(struct comp_dev *dev)
```

Comment:
this is interesting. this is what i wanted to achieve with the dummy codec :)

Reply:
We figured it would be smarter to just add it into the adapter so that codecs can be turned off and on even if codec did not add that functionality

Reply:
@cujomalainey sure. this is a very nice change.